core/controllers/editor_test.py

```diff
@@ -2937,3 +3016,127 @@ def test_rights_handler_returns_appropriate_rights(self):
         self.assertFalse(response['can_manage_voice_artist'])
 
         self.logout()
+
+
+class ImageUploadHandler(BaseEditorControllerTests):
+    """Test the handling of uploads by image uploads handler."""
+
+    def test_return_error_when_image_not_uploaded(self):
+        """Test error when no image is uploaded."""
+
+        self.login(self.EDITOR_EMAIL)
+        csrf_token = self.get_new_csrf_token()
+
+        exp_id = exp_fetchers.get_new_exploration_id()
+        self.save_new_valid_exploration(exp_id, self.editor_id)
+
+        filename = 'image_file.svg'
+        filename_prefix = 'image'
+
+        publish_url = '%s/%s/%s' % (
+            feconf.EXPLORATION_IMAGE_UPLOAD_PREFIX,
+            feconf.ENTITY_TYPE_EXPLORATION, exp_id)
+
+        response = self.post_json(
+            publish_url, {
+                'image': 'sample_image',
+                'filename': filename,
+                'filename_prefix': filename_prefix
+            },
+            csrf_token=csrf_token,
+            expected_status_int=400
```

Comment:
ditto: check that the upload actually doesn't happen. Please also do this for the rest of the tests even though I'm not going to leave a comment on each one

Reply:
I don't know if it is the best way to do it but my way goes something like, I checked if the uploaded file is present at the URL endpoint (or not) using AbstractFileSystem. 

Inside editor.py (lines 984-994), a similar thing was implemented to check if the uploaded file is already there or not. I used the same code to test it.